### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by bleafman**
- Added Auth0 wrapper to front end / top level
- Added PrivateRoute component to restrict access to private pages.

Screen Shot 2019-08-13 at 9 04 51 AM

### CODE REVIEW

File: client/src/Components/ErrorPage.jsx [hunk 1]
```diff
@@ -21,9 +21,7 @@ function ErrorPage() {
     <>
       <Grid item />
       <Grid item xs={12}>
-        <Paper className={classes.paper}>
-          Oh no! You&apos;ve found our 404 page!
-        </Paper>
+        <Paper className={classes.paper}>You found our 404 page!</Paper>
```
Comment (MitchellDill) [inline client/src/Components/ErrorPage.jsx]: What was once framed as a bad thing is now a beautiful discovery. The joy of finding a 404!

File: client/src/auth_config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const config = {
+  domain: 'greenocean.auth0.com',
+  clientId: '05RvxwAP7dSW5I9uPHxP6m7hVKHoIjS3'
```
Comment (MitchellDill) [inline client/src/auth_config.js]: Should this be in GitHub?

File: client/src/Components/ErrorPage.jsx (hunk 1 above)
Comment (bleafman) [reply client/src/Components/ErrorPage.jsx]: It throws weird errors about no &apos;

File: client/src/auth_config.js (hunk 2 above)
Comment (bleafman) [reply client/src/auth_config.js]: Nope! I added it to gitignore but the file extension was wrong so it was being included.

**Pull request merged**